cmd/cli/kubectl-kyverno/apply/apply_command_test.go

```diff
@@ -89,7 +89,7 @@ func Test_Apply(t *testing.T) {
 					Summary: preport.PolicyReportSummary{
 						Pass:  1,
 						Fail:  1,
-						Skip:  8,
+						Skip:  0,
```

Comment:
I believe Skip is 8 not zero due to the auto-gen rules.

We have 2 policies, each has 1 rule and 2 auto-gen rules and we have 2 resources. 
The 2 auto-gen rules are skipped for both resources so we have 4 skipped rules per policy.
 Therefore the total nu. of skipped rules is equal to 8.

Reply:
thanks @MariamFahmy98 , 
I will check it.

Reply:
True, when a resource and rule don't match it counts as a skip, this is a bug IMHO.

Reply:
@eddycharly @MariamFahmy98 Can you please check the logic that I have implemented for fixing the duplicate results counts.

Reply:
@hackeramitkumar can you explain your changes ?

Reply:
> @hackeramitkumar can you explain your changes ?

@eddycharly Currently we are processing the [VerifyImageResponse](https://github.com/kyverno/kyverno/blob/31bda44c5b341551335f77396b98fe8222f50e9d/cmd/cli/kubectl-kyverno/utils/common/kyverno_policies_types.go#L178C2-L182C3) and [validateImageResponse](https://github.com/kyverno/kyverno/blob/31bda44c5b341551335f77396b98fe8222f50e9d/cmd/cli/kubectl-kyverno/utils/common/kyverno_policies_types.go#L168C1-L173C1) in the kyverno_Policies_types.go .
And as a result, it updates the common.ResultCounts values here. And then we return this rc along with responses to the ApplyPolicyOnPath func(). And in that function, we are again updating the counts for the engine responses [here](https://github.com/kyverno/kyverno/blob/31bda44c5b341551335f77396b98fe8222f50e9d/cmd/cli/kubectl-kyverno/apply/apply_command.go#L402C4-L451C5) . 
So as a result we are counting few results twice. 

So we can remove duplicate logic from kyverno apply.  And we will update the resultsCounts in kyverno_policies_types only [here](https://github.com/kyverno/kyverno/blob/2ea25bdb76115bba29cb8fa4a37678e41051ddd1/cmd/cli/kubectl-kyverno/utils/common/common.go#L879C1-L918C2) and in apply_command.go and we will process the skipResponses only [here](https://github.com/kyverno/kyverno/blob/2ea25bdb76115bba29cb8fa4a37678e41051ddd1/cmd/cli/kubectl-kyverno/apply/apply_command.go#L480C1-L507C2) (Not update skip results counts because we have already done this in common.go )

Reply:
I need to look deeper, i don't know the CLI code well enough, will check later today and let you know.